packages/sandbox/src/lambda_function_log_streamer.ts

```diff
@@ -0,0 +1,169 @@
+    private readonly logsMonitor: CloudWatchLogEventMonitor,
+    private readonly backendOutputClient: BackendOutputClient,
+    private readonly printer: Printer
+  ) {}
+
+  public setOutputLocation = (outputLocation: string) => {
+    this.logsMonitor.setOutputLocation(outputLocation);
+  };
+
+  /**
+   * Starts watching logs in the given sandbox.
+   * @param sandboxBackendId The sandbox backend identifier.
+   * @param functionNamesToFilter The function names to filter.
+   */
+  public startWatchingLogs = async (
+    sandboxBackendId: BackendIdentifier,
+    functionNamesToFilter?: string[]
+  ) => {
+    const backendOutput: BackendOutput =
+      await this.backendOutputClient.getOutput(sandboxBackendId);
+
+    const definedFunctionsPayload =
+      backendOutput[functionOutputKey]?.payload.definedFunctions;
+    const deployedFunctionNames = definedFunctionsPayload
+      ? (JSON.parse(definedFunctionsPayload) as string[])
+      : [];
+
+    // To use list-tags API we need to convert function name to function Arn since it only accepts ARN as input
+    const deployedFunctionNameToArnMap = await this.getFunctionArnFromNames(
+      sandboxBackendId,
+      deployedFunctionNames
+    );
+
+    if (!deployedFunctionNameToArnMap) {
+      this.printer.log(
+        `[Sandbox] Could not find any function in stack ${BackendIdentifierConversions.toStackName(
+          sandboxBackendId
+        )}. Streaming function logs will be turned off.`,
+        LogLevel.DEBUG
+      );
+      return;
+    }
+
+    for (const entry of deployedFunctionNameToArnMap) {
+      const listTagsResponse = await this.lambda.send(
+        new ListTagsCommand({
+          Resource: entry.arn,
+        })
+      );
+      const friendlyFunctionName =
+        listTagsResponse.Tags?.[TagName.FRIENDLY_NAME];
+      if (!friendlyFunctionName) {
+        this.printer.log(
+          `[Sandbox] Could not find user defined name for lambda function ${entry.name}. Logs will not be streamed for this function.`,
+          LogLevel.DEBUG
+        );
+        continue;
+      }
+
+      if (
+        !functionNamesToFilter ||
+        functionNamesToFilter.includes(friendlyFunctionName)
+      ) {
+        this.logsMonitor?.addLogGroups(
+          friendlyFunctionName,
+          // a CW log group is implicitly created for each lambda function with the lambda function's name
+          `/aws/lambda/${entry.name}`
+        );
+        this.printer.log(
+          `[Sandbox] Logs for function ${friendlyFunctionName} will be streamed.`,
+          LogLevel.DEBUG
+        );
+      } else {
+        this.printer.log(
+          `[Sandbox] Skipping logs streaming for function ${friendlyFunctionName}. To stream logs for this function, provide its name in --function-name option`,
+          LogLevel.DEBUG
+        );
+      }
+    }
+
+    // finally start listening
+    this.logsMonitor?.activate();
+  };
+
+  public deactivate = () => {
```

Comment:
```suggestion
  public stopWatchingLogs = () => {
```
To match the other method and to avoid confusion with `CloudWatchLogEventMonitor.deactivate`